Microsoft.AVS.Management/Microsoft.AVS.Management.psm1

```diff
@@ -407,18 +450,53 @@ function New-LDAPSIdentitySource {
     else {
         Write-Host "No existing external identity sources found."
     }
-
     $Password = $Credential.GetNetworkCredential().Password
-    $DestinationFileArray = Get-Certificates -SSLCertificatesSasUrl $SSLCertificatesSasUrl -ErrorAction Stop
-    [System.Array]$Certificates =
-        foreach($CertFile in $DestinationFileArray) {
+    $DestinationFileArray = @()
+    if ($PSBoundParameters.ContainsKey('SSLCertificatesSasUrl')) {
+        $DestinationFileArray = Get-Certificates -SSLCertificatesSasUrl $SSLCertificatesSasUrl -ErrorAction Stop
+    } else {
+        $exportFolder = "$home/"
+        $remoteComputers = ,$PrimaryUrl
+        if ($PSBoundParameters.ContainsKey('SecondaryUrl')) { 
+            $remoteComputers += $SecondaryUrl 
+        }
+        
+        foreach ($computerUrl in $remoteComputers) {
             try {
-                [System.Security.Cryptography.X509Certificates.X509Certificate2]::CreateFromCertFile($certfile)
-            } catch {
-                Write-Error "Failure to convert file $certfile to a certificate $($PSItem.Exception.Message)"
-                throw "File to certificate conversion failed. See error message for more details"
+                if (![uri]::IsWellFormedUriString($computerUrl, 'Absolute')) { throw }
+                $ParsedUrl = [System.Uri]$computerUrl
             }
+            catch {
+                throw "Incorrect Url format entered from: $computerUrl"
+            }
+            if ($ParsedUrl.Host -match "^\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3}$" -and [bool]($ParsedUrl.Host -as [ipaddress])) { 
+                throw "Incorrect Url format. $computerUrl is an IP address. Consider changing it to an URL"
+            }
+            
+            $SSHOutput = Get-CertificateFromDomainController -ParsedUrl $ParsedUrl -computerUrl $computerUrl
+            
+            if ($SSHOutput -notmatch '(?s)(?<cert>-----BEGIN CERTIFICATE-----.*?-----END CERTIFICATE-----)') {
+                throw "The certificate from $computerUrl has an incorrect format"
+            } else {
+                $certs = select-string -inputobject $SSHOutput -pattern "(?s)(?<cert>-----BEGIN CERTIFICATE-----.*?-----END CERTIFICATE-----)" -allmatches
+                $cert = $certs.matches[0]
+                $exportPath = $exportFolder+($ParsedUrl.Host.split(".")[0])+".cer"
+                $cert.Value | Out-File $exportPath -Encoding ascii
+                $DestinationFileArray += $exportPath
+            }
+        }
+    }
+
+    [System.Array]$Certificates =
+    foreach($CertFile in $DestinationFileArray) {
+        try {
+            [System.Security.Cryptography.X509Certificates.X509Certificate2]::CreateFromCertFile($certfile)
+        } catch {
+            Write-Error "Failure to convert file $certfile to a certificate $($PSItem.Exception.Message)"
+            throw "File to certificate conversion failed. See error message for more details"
```

Comment:
Does this line ever run because I thought the default status of Write-Error (from agent) is --ErrorAction Stop

Reply:
Yeah, double-checked. Write-Error declares a non-terminating error. -ErrorAction Stop generates a terminating error.

Reply:
Yes, but while you're correct, I think the agent sets the default action to be -Stop so in effect this is running with -ErrorAction Stop